lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -0,0 +1,650 @@
+                sequence_number = sequence_number,
+                internal_log_rate_limit = true
+            );
+            return Ok(None);
+        }
+
+        message_state.add_chunk(sequence_number, chunk);
+
+        if let Some(message) = message_state.retrieve_message() {
+            state_lock.remove(&message_id);
+            Ok(Some(message))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Decode a GELF message that may be chunked or not. The source bytes are expected to be
+    /// datagram-based (or message-based), so it must not contain multiple GELF messages
+    /// delimited by '\0', such as it would be in a stream-based protocol.
+    pub fn decode_message(&mut self, mut src: Bytes) -> Result<Option<Bytes>, BoxedFramingError> {
+        let magic = src.get(0..2);
+        if magic.is_some_and(|magic| magic == GELF_MAGIC) {
+            src.advance(2);
+            let frame = self.decode_chunk(src);
+            return frame;
+        } else {
+            return Ok(Some(src));
+        }
+    }
+}
+
+impl Default for ChunkedGelfDecoder {
+    fn default() -> Self {
+        Self::new(DEFAULT_TIMEOUT_MILLIS, DEFAULT_PENDING_MESSAGES_LIMIT)
+    }
+}
+
+impl Decoder for ChunkedGelfDecoder {
+    type Item = Bytes;
+
+    type Error = BoxedFramingError;
+
+    fn decode(&mut self, src: &mut bytes::BytesMut) -> Result<Option<Self::Item>, Self::Error> {
+        if src.is_empty() {
+            return Ok(None);
+        }
+
+        // TODO: add a PR comment here stating that this will never call the decode_message since
+        // the bytes decoder will always return a Ok(None) in this method, but leaving this
+        // here for consistency. Would be better to add a unreachable/panic here if the inner decoder returns
+        // the Some variant?
+        self.bytes_decoder
+            .decode(src)?
+            .and_then(|frame| self.decode_message(frame).transpose())
+            .transpose()
+    }
+    fn decode_eof(&mut self, buf: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
+        if buf.is_empty() {
+            return Ok(None);
+        }
+
+        self.bytes_decoder
+            .decode_eof(buf)?
+            .and_then(|frame| self.decode_message(frame).transpose())
+            .transpose()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use bytes::{BufMut, BytesMut};
+    use rstest::{fixture, rstest};
+    use tracing_test::traced_test;
+
+    fn create_chunk(
+        message_id: u64,
+        sequence_number: u8,
+        total_chunks: u8,
+        payload: &str,
+    ) -> BytesMut {
+        let mut chunk = BytesMut::new();
+        chunk.put_slice(&GELF_MAGIC);
+        chunk.put_u64(message_id);
+        chunk.put_u8(sequence_number);
+        chunk.put_u8(total_chunks);
+        chunk.extend_from_slice(payload.as_bytes());
+        chunk
+    }
+
+    #[fixture]
+    fn unchunked_message() -> (BytesMut, String) {
```

Comment:
What is the correct spelling? I think its correct... should I add the word to the CI checklist?

Reply:
I think you can just add it to the spellcheck ignore list in `.github/actions/spelling/allow.txt`.